song_match/song_robot.py

```diff
@@ -48,11 +49,56 @@ def __init__(self, robot: Robot, song: Song):
         await note_cube.blink_and_play_note()
         await self.__wait_until_animation_finished()
 
+    async def play_note_with_error(self, note: Note, round: int):
+        """Make Cozmo play a note.
+
+        :param note: The :class:`~song_match.song.note.Note` to play.
+        :return: None
+        """
+        difficulty = .99
+        rand = random()
+
+        round_difficulty = difficulty ** round
+
+        cube_id = self._song.get_cube_id(note)
+
+        if round_difficulty < rand:
+            cozmoError = True
+            if cube_id == 1:
+                cube_id = 2
+            elif cube_id == 2:
+                cube_id = 3
+            else:
+                cube_id = 1
+        else:
+            cozmoError = True
+        note_cube = self.__get_note_cube(cube_id)
+        self.__tap_cube(cube_id)
+        await sleep(self._NOTE_DELAY)
+        await note_cube.blink_and_play_note()
+        await self.__wait_until_animation_finished()
+        return cozmoError
+
+    async def play_note_notap(self, note: Note) -> None:
```

Comment:
I think you should move this to the `SongMatch` class. Then you could rename it to `play_note` and have it make sense.

Reply:
I understand this logic in regards to our chat after we left the library. It should definitely be changed.